ladon.go

```diff
@@ -113,10 +140,18 @@ func (l *Ladon) DoPoliciesAllow(r *Request, policies []Policy) (err error) {
 	}
 
 	if !allowed {
+		if l.Metric != nil {
+			go l.Metric.RequestNoMatch(*r)
+		}
+
 		l.auditLogger().LogRejectedAccessRequest(r, policies, deciders)
 		return errors.WithStack(ErrRequestDenied)
 	}
 
+	if l.Metric != nil {
+		l.Metric.RequestAllowedBy(*r, deciders)
```

Comment:
One last thing! You could do it similar to `auditlogger()` instead - so `l.metric().RequestAllowedBy ...` and then get rid of all the `if l.Metric != nil` by simply checking that in `metric()`

Reply:
Really like that approach - just implemented it and also moved the code similar to what we have with `auditlogger` 👍